### PR TITLE
Drop json_parse_exception messages for bulk failures

### PR DESCRIPTION
**Pull request by jcantrill**
When Elasticsearch rejects a record because it is unable to parse the content(i.e due to an encoding issue), the record blocks resubmission of the remaining records since the specific record can never be parsed by the server.  

This PR:
* Add functionality to drop specific status:500 records which are not recoverable

Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1951605

(check all that apply)
- [x]  tests added
- [x] tests passing
- [ ] README updated (if needed)
- [ ] README Table of Contents updated (if needed)
- [ ] History.md and `version` in gemspec are untouched
- [x] backward compatible
- [ ] feature works in `elasticsearch_dynamic` (not required but recommended)

### CODE REVIEW

File: lib/fluent/plugin/elasticsearch_error_handler.rb [hunk 1]
```diff
@@ -23,6 +23,10 @@ def unrecoverable_error?(type)
     unrecoverable_error_types.include?(type)
   end
 
+  def unrecoverable_record_error?(type)
+    ['json_parse_exception'].include?(type)
```
Comment (cosmo0920) [inline lib/fluent/plugin/elasticsearch_error_handler.rb:27]: It's pretty good for adding unrecoverable record error concept, but I think that `es_rejected_execution_exception` is also unrecoverable exception.

What do you think about this?
Keep as-is for now is also OK for me.

**Comment by cosmo0920** [comment]
Your PR concept is pretty good for me.
I added a comment for small concerns.

File: lib/fluent/plugin/elasticsearch_error_handler.rb (hunk 1 above)
Comment (jcantrill) [reply lib/fluent/plugin/elasticsearch_error_handler.rb:27]: I defer to you here as I do not know.  If this is an exception one might expect from trying to push an individual record and it is not possible to recover or not possible to successfully resubmit, I agree we can add this to the list

**Comment by jcantrill** [comment]
> Your PR concept is pretty good for me.
> I added a comment for small concerns.

Please advise if you wish for me to make changes before merging

**Comment by jcantrill** [comment]
@cosmo0920 ... ping.. please advise

**Comment by cosmo0920** [comment]
Sorry to be late merging. I decided this PR merge as-is.

**Pull request merged**